website/docs/r/advanced_cluster.html.markdown

```diff
@@ -287,6 +287,15 @@ replication_specs {
 
 * `disk_gb_enabled` - (Optional) Flag that indicates whether this cluster enables disk auto-scaling. This parameter defaults to true.
 * `compute_enabled` - (Optional) Flag that indicates whether instance size auto-scaling is enabled. This parameter defaults to false.
+
+~> **IMPORTANT:** If `compute_enabled` is true, then Atlas will automatically scale up to the maximum provided and down to the minimum, if provided.
+This will cause the value of `instance_size` returned to potential be different than what is specified in the Terraform config and if one then applies a plan, not noting this, Terraform will scale the cluster back down to the original `instance_size` value.
+To prevent this a lifecycle customization should be used, i.e.:  
+`lifecycle {
+  ignore_changes = [instance_size]
```

Comment:
@Zuhairahmed @martinstibbe Hi,
Maybe I'm wrong, but I can't include a nested attribute into `ignore_changes`
For example,
```HCL
#...
  lifecycle {
    # as autoscaling is enabled
    ignore_changes = [
      disk_size_gb,
      instance_size
    ]
  }
```
`terraform plan`
```
╷
│ Error: Unsupported attribute
│ 
│   on mongodb_atlas_cluster.tf line 72, in resource "mongodbatlas_advanced_cluster" "product":
│   72:       instance_size
│ 
│ This object has no argument, nested block, or exported attribute named "instance_size".
```

```
✗ terraform version
Terraform v1.2.9
on darwin_arm64

+ provider registry.terraform.io/mongodb/mongodbatlas v1.5.0
```

More details - https://github.com/mongodb/terraform-provider-mongodbatlas/issues/888

Reply:
thanks @ivan-sukhomlyn, we are actively working on this issue and should have release out as part of v1.7 release in a few weeks time. Internal ticket [INTMDB-464](https://jira.mongodb.org/browse/INTMDB-464)